Account UI E2E / personal-info/personal-info.spec.ts is unstable

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

account/ui

### Describe the bug

From a run over night with retries disabled the test `personal-info/personal-info.spec.ts` failed 100 out of 100 runs.

Example of full log is available at https://github.com/stianst/keycloak/actions/runs/11955347294/job/33332680767 

Extract from log:
```
2024-11-21T19:30:37.4230576Z Running 26 tests using 1 worker
2024-11-21T19:30:37.9080969Z (node:5557) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time
2024-11-21T19:30:37.9082309Z (Use `node --trace-warnings ...` to show where the warning was created)
2024-11-21T19:30:48.6205992Z (node:5571) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time
2024-11-21T19:30:48.6207547Z (Use `node --trace-warnings ...` to show where the warning was created)
2024-11-21T19:31:08.2057629Z NetworkError: HTTP 401 Unauthorized
2024-11-21T19:31:08.2058903Z     at fetchWithError (file:///home/runner/work/keycloak/keycloak/js/libs/keycloak-admin-client/lib/utils/fetchWithError.js:16:15)
2024-11-21T19:31:08.2060666Z [90m    at processTicksAndRejections (node:internal/process/task_queues:95:5)[39m
2024-11-21T19:31:08.2062060Z     at Agent.#requestWithParams (file:///home/runner/work/keycloak/keycloak/js/libs/keycloak-admin-client/lib/resources/agent.js:127:25)
2024-11-21T19:31:08.2063840Z     at createUser [90m(file:///home/runner/work/keycloak/keycloak/js/apps/account-ui/[39mtest/admin-client.ts:98:5[90m)[39m
2024-11-21T19:31:08.2065507Z     at inRealm [90m(file:///home/runner/work/keycloak/keycloak/js/apps/account-ui/[39mtest/admin-client.ts:37:12[90m)[39m
2024-11-21T19:31:08.2067191Z     at [90mfile:///home/runner/work/keycloak/keycloak/js/apps/account-ui/[39mtest/personal-info/personal-info.spec.ts:21:7
2024-11-21T19:31:08.2069103Z     at /home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/fixtureRunner.js:250:7
2024-11-21T19:31:08.2071095Z     at /home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/testInfo.js:286:11
2024-11-21T19:31:08.2073380Z     at TimeoutManager.withRunnable (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/timeoutManager.js:64:14)
2024-11-21T19:31:08.2075793Z     at TestInfoImpl._runAsStage (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/testInfo.js:284:7)
2024-11-21T19:31:08.2078413Z     at FixtureRunner.resolveParametersAndRunFunction (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/fixtureRunner.js:246:5)
2024-11-21T19:31:08.2081171Z     at /home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/workerMain.js:535:13
2024-11-21T19:31:08.2083181Z     at /home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/testInfo.js:286:11
2024-11-21T19:31:08.2085454Z     at TimeoutManager.withRunnable (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/timeoutManager.js:52:27)
2024-11-21T19:31:08.2087863Z     at TestInfoImpl._runAsStage (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/testInfo.js:284:7)
2024-11-21T19:31:08.2090686Z     at WorkerMain._runAllHooksForSuite (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/workerMain.js:516:9)
2024-11-21T19:31:08.2093236Z     at WorkerMain._runBeforeAllHooksForSuite (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/workerMain.js:509:5)
2024-11-21T19:31:08.2095520Z     at /home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/workerMain.js:316:37
2024-11-21T19:31:08.2097672Z     at /home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/testInfo.js:286:11
2024-11-21T19:31:08.2104785Z     at TimeoutManager.withRunnable (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/timeoutManager.js:52:27)
2024-11-21T19:31:08.2107586Z     at TestInfoImpl._runAsStage (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/testInfo.js:284:7)
2024-11-21T19:31:08.2109727Z     at /home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/workerMain.js:309:7
2024-11-21T19:31:08.2112200Z     at /home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/testInfo.js:286:11
2024-11-21T19:31:08.2114317Z     at TimeoutManager.withRunnable (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/timeoutManager.js:52:27)
2024-11-21T19:31:08.2116323Z     at TestInfoImpl._runAsStage (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/testInfo.js:284:7)
2024-11-21T19:31:08.2117679Z     at WorkerMain._runTest (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/workerMain.js:281:5)
2024-11-21T19:31:08.2118974Z     at WorkerMain.runTestGroup (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/workerMain.js:205:11)
2024-11-21T19:31:08.2120317Z     at process.<anonymous> (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/common/process.js:95:22) {
2024-11-21T19:31:08.2120970Z   response: Response {
2024-11-21T19:31:08.2121215Z     status: [33m401[39m,
2024-11-21T19:31:08.2121512Z     statusText: [32m'Unauthorized'[39m,
2024-11-21T19:31:08.2121768Z     headers: Headers {
2024-11-21T19:31:08.2122063Z       [32m'content-length'[39m: [32m'33'[39m,
2024-11-21T19:31:08.2122457Z       [32m'content-type'[39m: [32m'application/json'[39m,
2024-11-21T19:31:08.2122880Z       [32m'referrer-policy'[39m: [32m'no-referrer'[39m,
2024-11-21T19:31:08.2123418Z       [32m'strict-transport-security'[39m: [32m'max-age=31536000; includeSubDomains'[39m,
2024-11-21T19:31:08.2123935Z       [32m'x-content-type-options'[39m: [32m'nosniff'[39m,
2024-11-21T19:31:08.2124333Z       [32m'x-frame-options'[39m: [32m'SAMEORIGIN'[39m,
2024-11-21T19:31:08.2124965Z       [32m'x-xss-protection'[39m: [32m'1; mode=block'[39m
2024-11-21T19:31:08.2125247Z     },
2024-11-21T19:31:08.2125739Z     body: ReadableStream { locked: [33mtrue[39m, state: [32m'closed'[39m, supportsBYOB: [33mtrue[39m },
2024-11-21T19:31:08.2126191Z     bodyUsed: [33mtrue[39m,
2024-11-21T19:31:08.2126435Z     ok: [33mfalse[39m,
2024-11-21T19:31:08.2126695Z     redirected: [33mfalse[39m,
2024-11-21T19:31:08.2126955Z     type: [32m'basic'[39m,
2024-11-21T19:31:08.2127377Z     url: [32m'http://localhost:8080/admin/realms/user-profile/users'[39m
2024-11-21T19:31:08.2127694Z   },
2024-11-21T19:31:08.2128003Z   responseData: { error: [32m'HTTP 401 Unauthorized'[39m }
2024-11-21T19:31:08.2128301Z }
2024-11-21T19:31:38.2274224Z NetworkError: HTTP 401 Unauthorized
2024-11-21T19:31:38.2276703Z     at fetchWithError (file:///home/runner/work/keycloak/keycloak/js/libs/keycloak-admin-client/lib/utils/fetchWithError.js:16:15)
2024-11-21T19:31:38.2278768Z [90m    at processTicksAndRejections (node:internal/process/task_queues:95:5)[39m
2024-11-21T19:31:38.2305005Z     at Agent.#requestWithParams (file:///home/runner/work/keycloak/keycloak/js/libs/keycloak-admin-client/lib/resources/agent.js:127:25)
2024-11-21T19:31:38.2307961Z     at deleteUser [90m(file:///home/runner/work/keycloak/keycloak/js/apps/account-ui/[39mtest/admin-client.ts:129:19[90m)[39m
2024-11-21T19:31:38.2309675Z     at inRealm [90m(file:///home/runner/work/keycloak/keycloak/js/apps/account-ui/[39mtest/admin-client.ts:37:12[90m)[39m
2024-11-21T19:31:38.2311622Z     at [90mfile:///home/runner/work/keycloak/keycloak/js/apps/account-ui/[39mtest/personal-info/personal-info.spec.ts:23:29
2024-11-21T19:31:38.2314158Z     at /home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/fixtureRunner.js:250:7
2024-11-21T19:31:38.2316186Z     at /home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/testInfo.js:286:11
2024-11-21T19:31:38.2318466Z     at TimeoutManager.withRunnable (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/timeoutManager.js:64:14)
2024-11-21T19:31:38.2320833Z     at TestInfoImpl._runAsStage (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/testInfo.js:284:7)
2024-11-21T19:31:38.2323318Z     at FixtureRunner.resolveParametersAndRunFunction (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/fixtureRunner.js:246:5)
2024-11-21T19:31:38.2325466Z     at /home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/workerMain.js:535:13
2024-11-21T19:31:38.2327304Z     at /home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/testInfo.js:286:11
2024-11-21T19:31:38.2329455Z     at TimeoutManager.withRunnable (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/timeoutManager.js:52:27)
2024-11-21T19:31:38.2332833Z     at TestInfoImpl._runAsStage (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/testInfo.js:284:7)
2024-11-21T19:31:38.2334549Z     at WorkerMain._runAllHooksForSuite (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/workerMain.js:516:9)
2024-11-21T19:31:38.2337978Z     at WorkerMain._runAfterAllHooksForSuite (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/workerMain.js:565:5)
2024-11-21T19:31:38.2340415Z     at /home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/workerMain.js:403:13
2024-11-21T19:31:38.2342372Z     at /home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/testInfo.js:286:11
2024-11-21T19:31:38.2344273Z     at TimeoutManager.withRunnable (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/timeoutManager.js:52:27)
2024-11-21T19:31:38.2345923Z     at TestInfoImpl._runAsStage (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/testInfo.js:284:7)
2024-11-21T19:31:38.2347218Z     at WorkerMain._runTest (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/workerMain.js:355:5)
2024-11-21T19:31:38.2348513Z     at WorkerMain.runTestGroup (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/worker/workerMain.js:205:11)
2024-11-21T19:31:38.2349806Z     at process.<anonymous> (/home/runner/work/keycloak/keycloak/node_modules/[4m.pnpm[24m/playwright@1.49.0/node_modules/[4mplaywright[24m/lib/common/process.js:95:22) {
2024-11-21T19:31:38.2350447Z   response: Response {
2024-11-21T19:31:38.2350691Z     status: [33m401[39m,
2024-11-21T19:31:38.2350974Z     statusText: [32m'Unauthorized'[39m,
2024-11-21T19:31:38.2351224Z     headers: Headers {
2024-11-21T19:31:38.2351520Z       [32m'content-length'[39m: [32m'33'[39m,
2024-11-21T19:31:38.2351922Z       [32m'content-type'[39m: [32m'application/json'[39m,
2024-11-21T19:31:38.2352577Z       [32m'referrer-policy'[39m: [32m'no-referrer'[39m,
2024-11-21T19:31:38.2353114Z       [32m'strict-transport-security'[39m: [32m'max-age=31536000; includeSubDomains'[39m,
2024-11-21T19:31:38.2353635Z       [32m'x-content-type-options'[39m: [32m'nosniff'[39m,
2024-11-21T19:31:38.2354032Z       [32m'x-frame-options'[39m: [32m'SAMEORIGIN'[39m,
2024-11-21T19:31:38.2354427Z       [32m'x-xss-protection'[39m: [32m'1; mode=block'[39m
2024-11-21T19:31:38.2354698Z     },
2024-11-21T19:31:38.2355184Z     body: ReadableStream { locked: [33mtrue[39m, state: [32m'closed'[39m, supportsBYOB: [33mtrue[39m },
2024-11-21T19:31:38.2355636Z     bodyUsed: [33mtrue[39m,
2024-11-21T19:31:38.2356102Z     ok: [33mfalse[39m,
2024-11-21T19:31:38.2356365Z     redirected: [33mfalse[39m,
2024-11-21T19:31:38.2356622Z     type: [32m'basic'[39m,
2024-11-21T19:31:38.2357258Z     url: [32m'http://localhost:8080/admin/realms/user-profile/users?username=user-de20dd76-4103-4559-b4f7-a17bf838a45b'[39m
2024-11-21T19:31:38.2357763Z   },
2024-11-21T19:31:38.2358075Z   responseData: { error: [32m'HTTP 401 Unauthorized'[39m }
2024-11-21T19:31:38.2358386Z }
2024-11-21T19:31:38.7638777Z (node:5923) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time
2024-11-21T19:31:38.7640473Z (Use `node --trace-warnings ...` to show where the warning was created)
2024-11-21T19:31:48.3515511Z ················°T········
2024-11-21T19:31:48.3521424Z 
2024-11-21T19:31:48.3531002Z   1) [chromium] › personal-info/personal-info.spec.ts:25:3 › Personal info page › sets basic information 
2024-11-21T19:31:48.3531468Z 
2024-11-21T19:31:48.3531698Z     [31mTest timeout of 30000ms exceeded.[39m
2024-11-21T19:31:48.3531916Z 
2024-11-21T19:31:48.3532110Z     Error: locator.fill: Test timeout of 30000ms exceeded.
2024-11-21T19:31:48.3532423Z     Call log:
2024-11-21T19:31:48.3532734Z     [2m  - waiting for getByTestId('email')[22m
2024-11-21T19:31:48.3532991Z 
2024-11-21T19:31:48.3532997Z 
2024-11-21T19:31:48.3533149Z       26 |     await login(page, user, "pwd", realm);
2024-11-21T19:31:48.3533433Z       27 |
2024-11-21T19:31:48.3533938Z     > 28 |     await page.getByTestId("email").fill(`${user}@somewhere.com`);
2024-11-21T19:31:48.3534557Z          |                                     ^
2024-11-21T19:31:48.3535499Z       29 |     await page.getByTestId("firstName").fill("Erik");
2024-11-21T19:31:48.3536246Z       30 |     await page.getByTestId("lastName").fill("de Wit");
2024-11-21T19:31:48.3536658Z       31 |     await page.getByTestId("save").click();
2024-11-21T19:31:48.3537313Z         at /home/runner/work/keycloak/keycloak/js/apps/account-ui/test/personal-info/personal-info.spec.ts:28:37
2024-11-21T19:31:48.3537781Z 
2024-11-21T19:31:48.3537866Z   1 failed
2024-11-21T19:31:48.3538448Z     [chromium] › personal-info/personal-info.spec.ts:25:3 › Personal info page › sets basic information 
2024-11-21T19:31:48.3538925Z   1 skipped
2024-11-21T19:31:48.3539128Z   24 passed (1.2m)
2024-11-21T19:31:48.3571353Z ##[error]  1) [chromium] › personal-info/personal-info.spec.ts:25:3 › Personal info page › sets basic information 
    Test timeout of 30000ms exceeded.
2024-11-21T19:31:48.3589726Z ##[error]  1) [chromium] › personal-info/personal-info.spec.ts:25:3 › Personal info page › sets basic information 
    Error: locator.fill: Test timeout of 30000ms exceeded.
    Call log:
      - waiting for getByTestId('email')


      26 |     await login(page, user, "pwd", realm);
      27 |
    > 28 |     await page.getByTestId("email").fill(`${user}@somewhere.com`);
         |                                     ^
      29 |     await page.getByTestId("firstName").fill("Erik");
      30 |     await page.getByTestId("lastName").fill("de Wit");
      31 |     await page.getByTestId("save").click();
        at /home/runner/work/keycloak/keycloak/js/apps/account-ui/test/personal-info/personal-info.spec.ts:28:37
2024-11-21T19:31:48.3597084Z ##[notice]  1 failed
    [chromium] › personal-info/personal-info.spec.ts:25:3 › Personal info page › sets basic information 
  1 skipped
  24 passed (1.2m)
2024-11-21T19:31:48.4000026Z 
2024-11-21T19:31:48.4000849Z ❌ [test] exited with exit code 1.
2024-11-21T19:31:48.4004630Z ❌ 1 script failed.
```



### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

Test passes

### Actual behavior

Test fails

### How to Reproduce?

run test on GitHub Actions with retries disabled (https://github.com/stianst/keycloak/commit/8bc6557cf96d2b43eb4251de75c5afb6e57f44f7)

### Anything else?

_No response_

--- a/js/apps/account-ui/test/account-security/linked-accounts.spec.ts
+++ b/js/apps/account-ui/test/account-security/linked-accounts.spec.ts
@@ -11,7 +11,6 @@ import {
   deleteIdentityProvider,
   deleteUser,
   findClientByClientId,
-  inRealm,
 } from "../admin-client";
 import { SERVER_URL } from "../constants";
 import groupsIdPClient from "../realms/groups-idp.json" assert { type: "json" };
@@ -23,7 +22,11 @@ test.describe("Account linking", () => {
   let user: string;
   // Tests for keycloak account console, section Account linking in Account security
   test.beforeAll(async () => {
-    user = await createRandomUserWithPassword("user-" + randomUUID(), "pwd");
+    user = await createRandomUserWithPassword(
+      "user-" + randomUUID(),
+      "pwd",
+      realm,
+    );
 
     const kcGroupsIdpId = await findClientByClientId("groups-idp");
     if (kcGroupsIdpId) {
@@ -49,17 +52,17 @@ test.describe("Account linking", () => {
       },
     };
 
-    await inRealm(realm, () => createIdentityProvider(idp));
+    await createIdentityProvider(idp, realm);
   });
 
   test.afterAll(async () => {
-    await deleteUser(user);
+    await deleteUser(user, realm);
   });
   test.afterAll(async () => {
     await deleteClient(groupIdPClientId);
   });
   test.afterAll(async () => {
-    await inRealm(realm, () => deleteIdentityProvider("master-idp"));
+    await deleteIdentityProvider("master-idp", realm);
   });
 
   test("Linking", async ({ page }) => {

--- a/js/apps/account-ui/test/admin-client.ts
+++ b/js/apps/account-ui/test/admin-client.ts
@@ -30,16 +30,6 @@ export async function useTheme() {
   );
 }
 
-export async function inRealm<T>(realm: string, fn: () => Promise<T>) {
-  const prevRealm = adminClient.realmName;
-  adminClient.realmName = realm;
-  try {
-    return await fn();
-  } finally {
-    adminClient.realmName = prevRealm;
-  }
-}
-
 export async function importRealm(realm: RealmRepresentation) {
   await adminClient.realms.create(realm);
 }
@@ -66,12 +56,16 @@ export async function deleteClient(id: string) {
 
 export async function createIdentityProvider(
   idp: IdentityProviderRepresentation,
+  realm = DEFAULT_REALM,
 ): Promise<string> {
-  return adminClient.identityProviders.create(idp)["id"];
+  return adminClient.identityProviders.create({ ...idp, realm })["id"];
 }
 
-export async function deleteIdentityProvider(alias: string) {
-  await adminClient.identityProviders.del({ alias });
+export async function deleteIdentityProvider(
+  alias: string,
+  realm = DEFAULT_REALM,
+) {
+  await adminClient.identityProviders.del({ alias, realm });
 }
 
 export async function importUserProfile(
@@ -93,9 +87,12 @@ export async function enableLocalization(realm = DEFAULT_REALM) {
   );
 }
 
-export async function createUser(user: UserRepresentation) {
+export async function createUser(
+  user: UserRepresentation,
+  realm = DEFAULT_REALM,
+) {
   try {
-    await adminClient.users.create(user);
+    await adminClient.users.create({ ...user, realm });
   } catch (error) {
     console.error(error);
   }
@@ -104,19 +101,29 @@ export async function createUser(user: UserRepresentation) {
 export async function createRandomUserWithPassword(
   username: string,
   password: string,
+  realm: string,
   props?: UserRepresentation,
 ): Promise<string> {
-  return createUser({
-    username: username,
-    enabled: true,
-    credentials: [
-      {
-        type: "password",
-        value: password,
-      },
-    ],
-    ...props,
-  }).then(() => username);
+  await adminClient.auth({
+    username: "admin",
+    password: "admin",
+    grantType: "password",
+    clientId: "admin-cli",
+  });
+  return createUser(
+    {
+      username: username,
+      enabled: true,
+      credentials: [
+        {
+          type: "password",
+          value: password,
+        },
+      ],
+      ...props,
+    },
+    realm,
+  ).then(() => username);
 }
 
 export async function getUserByUsername(username: string, realm: string) {
@@ -124,11 +131,15 @@ export async function getUserByUsername(username: string, realm: string) {
   return users.length > 0 ? users[0] : undefined;
 }
 
-export async function deleteUser(username: string) {
+export async function deleteUser(username: string, realm = DEFAULT_REALM) {
   try {
-    const users = await adminClient.users.find({ username });
+    const users = await adminClient.users.find({ username, realm });
+    if (users.length === 0) {
+      console.warn(`User ${username} not found in realm ${realm}`);
+      return;
+    }
     const { id } = users[0];
-    await adminClient.users.del({ id: id! });
+    await adminClient.users.del({ id: id!, realm });
   } catch (error) {
     console.error(error);
   }

--- a/js/apps/account-ui/test/personal-info/personal-info.spec.ts
+++ b/js/apps/account-ui/test/personal-info/personal-info.spec.ts
@@ -5,7 +5,6 @@ import {
   deleteUser,
   enableLocalization,
   importUserProfile,
-  inRealm,
 } from "../admin-client";
 import { login } from "../login";
 import userProfileConfig from "./user-profile.json" assert { type: "json" };
@@ -16,11 +15,8 @@ const realm = "user-profile";
 test.describe("Personal info page", () => {
   const user = "user-" + randomUUID();
 
-  test.beforeAll(
-    async () =>
-      await inRealm(realm, () => createRandomUserWithPassword(user, "pwd")),
-  );
-  test.afterAll(async () => await inRealm(realm, () => deleteUser(user)));
+  test.beforeAll(() => createRandomUserWithPassword(user, "pwd", realm));
+  test.afterAll(async () => deleteUser(user, realm));
 
   test("sets basic information", async ({ page }) => {
     await login(page, user, "pwd", realm);
@@ -39,8 +35,11 @@ test.describe("Personal info with userprofile enabled", () => {
   let user: string;
   test.beforeAll(async () => {
     await importUserProfile(userProfileConfig as UserProfileConfig, realm);
-    user = await inRealm(realm, () =>
-      createRandomUserWithPassword("user-" + randomUUID(), "jdoe", {
+    user = await createRandomUserWithPassword(
+      "user-" + randomUUID(),
+      "jdoe",
+      realm,
+      {
         email: "jdoe@keycloak.org",
         firstName: "John",
         lastName: "Doe",
@@ -48,11 +47,11 @@ test.describe("Personal info with userprofile enabled", () => {
         clientRoles: {
           account: ["manage-account"],
         },
-      }),
+      },
     );
   });
 
-  test.afterAll(async () => await inRealm(realm, () => deleteUser(user)));
+  test.afterAll(() => deleteUser(user, realm));
 
   test("render user profile fields", async ({ page }) => {
     await login(page, user, "jdoe", realm);
@@ -121,8 +120,10 @@ test.describe("Personal info with userprofile enabled", () => {
 test.describe("Realm localization", () => {
   test.beforeAll(() => enableLocalization());
   test("change locale", async ({ page }) => {
-    const user = await inRealm(realm, () =>
-      createRandomUserWithPassword("user-" + randomUUID(), "pwd"),
+    const user = await createRandomUserWithPassword(
+      "user-" + randomUUID(),
+      "pwd",
+      realm,
     );
 
     await login(page, user, "pwd", realm);

--- a/js/apps/account-ui/test/realms/groups-realm.json
+++ b/js/apps/account-ui/test/realms/groups-realm.json
@@ -105,53 +105,6 @@
     }
   ],
   "clients": [
-    {
-      "clientId": "security-admin-console-v2",
-      "rootUrl": "http://localhost:8080/",
-      "adminUrl": "http://localhost:8080/",
-      "surrogateAuthRequired": false,
-      "enabled": true,
-      "alwaysDisplayInConsole": false,
-      "clientAuthenticatorType": "client-secret",
-      "redirectUris": [
-          "http://localhost:8080/*"
-      ],
-      "webOrigins": [
-          "http://localhost:8080"
-      ],
-      "notBefore": 0,
-      "bearerOnly": false,
-      "consentRequired": false,
-      "standardFlowEnabled": true,
-      "implicitFlowEnabled": false,
-      "directAccessGrantsEnabled": true,
-      "serviceAccountsEnabled": false,
-      "publicClient": true,
-      "frontchannelLogout": false,
-      "protocol": "openid-connect",
-      "attributes": {},
-      "authenticationFlowBindingOverrides": {},
-      "fullScopeAllowed": true,
-      "nodeReRegistrationTimeout": -1,
-      "defaultClientScopes": [
-          "web-origins",
-          "role_list",
-          "roles",
-          "profile",
-          "email"
-      ],
-      "optionalClientScopes": [
-          "address",
-          "phone",
-          "offline_access",
-          "microprofile-jwt"
-      ],
-      "access": {
-          "view": true,
-          "configure": true,
-          "manage": true
-      }
-    }
   ],
   "identityProviders": [
     {

--- a/js/apps/account-ui/test/realms/resources-realm.json
+++ b/js/apps/account-ui/test/realms/resources-realm.json
@@ -191,53 +191,6 @@
         ],
         "decisionStrategy": "UNANIMOUS"
       }
-    },
-    {
-      "clientId": "security-admin-console-v2",
-      "rootUrl": "http://localhost:8080/",
-      "adminUrl": "http://localhost:8080/",
-      "surrogateAuthRequired": false,
-      "enabled": true,
-      "alwaysDisplayInConsole": false,
-      "clientAuthenticatorType": "client-secret",
-      "redirectUris": [
-          "http://localhost:8080/*"
-      ],
-      "webOrigins": [
-          "http://localhost:8080"
-      ],
-      "notBefore": 0,
-      "bearerOnly": false,
-      "consentRequired": false,
-      "standardFlowEnabled": true,
-      "implicitFlowEnabled": false,
-      "directAccessGrantsEnabled": true,
-      "serviceAccountsEnabled": false,
-      "publicClient": true,
-      "frontchannelLogout": false,
-      "protocol": "openid-connect",
-      "attributes": {},
-      "authenticationFlowBindingOverrides": {},
-      "fullScopeAllowed": true,
-      "nodeReRegistrationTimeout": -1,
-      "defaultClientScopes": [
-          "web-origins",
-          "role_list",
-          "roles",
-          "profile",
-          "email"
-      ],
-      "optionalClientScopes": [
-          "address",
-          "phone",
-          "offline_access",
-          "microprofile-jwt"
-      ],
-      "access": {
-          "view": true,
-          "configure": true,
-          "manage": true
-      }
     }
   ]
 }

--- a/js/apps/account-ui/test/realms/user-profile-realm.json
+++ b/js/apps/account-ui/test/realms/user-profile-realm.json
@@ -103,49 +103,6 @@
     }
   ],
   "clients": [
-    {
-      "clientId": "security-admin-console-v2",
-      "rootUrl": "http://localhost:8080/",
-      "adminUrl": "http://localhost:8080/",
-      "surrogateAuthRequired": false,
-      "enabled": true,
-      "alwaysDisplayInConsole": false,
-      "clientAuthenticatorType": "client-secret",
-      "redirectUris": ["http://localhost:8080/*"],
-      "webOrigins": ["http://localhost:8080"],
-      "notBefore": 0,
-      "bearerOnly": false,
-      "consentRequired": false,
-      "standardFlowEnabled": true,
-      "implicitFlowEnabled": false,
-      "directAccessGrantsEnabled": true,
-      "serviceAccountsEnabled": false,
-      "publicClient": true,
-      "frontchannelLogout": false,
-      "protocol": "openid-connect",
-      "attributes": {},
-      "authenticationFlowBindingOverrides": {},
-      "fullScopeAllowed": true,
-      "nodeReRegistrationTimeout": -1,
-      "defaultClientScopes": [
-        "web-origins",
-        "role_list",
-        "roles",
-        "profile",
-        "email"
-      ],
-      "optionalClientScopes": [
-        "address",
-        "phone",
-        "offline_access",
-        "microprofile-jwt"
-      ],
-      "access": {
-        "view": true,
-        "configure": true,
-        "manage": true
-      }
-    }
   ],
   "users": [
     {

